packages/form-js-viewer/src/features/expression-language/FeelersTemplating.js

```diff
@@ -1,11 +1,49 @@
-import { evaluate as evaluateFeelers } from 'feelers';
+import { parser as feelersParser, buildSimpleTree, evaluate as evaluateFeelers } from 'feelers';
 import { isString } from 'min-dash';
+import { getFlavouredFeelVariableNames } from './variableExtractionHelpers';
 
 export default class FeelersTemplating {
 
   constructor() { }
 
-  isTemplate(value) { return isString(value) && (value.startsWith('=') || /{{/.test(value)); }
+  /**
+   * Determines if the given value is a feelers template.
+   *
+   * @param {any} value
+   * @returns {boolean}
+   *
+   */
+  isTemplate(value) { return isString(value) && (value.startsWith('=') || /{{.*?}}/.test(value)); }
+
+
+  /**
+   * Retrieve variable names from a given feelers template.
+   *
+   * @param {string} template
+   *
+   * @returns {string[]}
+   */
+  getVariableNames(template) {
+
+    if (!this.isTemplate(template)) {
+      return [];
+    }
+
+    const expressions = this._extractExpressionsWithDepth(template);
+
+    // defines special accessors, and the change(s) in depth they could imply (e.g. parent can be used to access the parent context (depth - 1) or a child variable named parent (depth + 1)
+    const specialDepthAccessors = {
+      parent: [ -1, 1 ],
+      _parent_: [ -1 ],
+      this: [ 0, 1 ],
+      _this_: [ 0 ],
+    };
+
+    return expressions.reduce((variables, { expression, depth }) => {
+      return [ ...variables, ...getFlavouredFeelVariableNames(expression, 'expression', { depth, specialDepthAccessors }) ];
+    }, []);
+  }
```

Comment:
How frequently is called? and how big are the datasets usually?

Might be better to reconsider using spread inside a loop for performance reasons

Reply:
Anywhere from 1 - 20 I'd say. But I'll do a concat instead here.